node-graph/gcore/src/vector/vector_data/indexed.rs

```diff
@@ -0,0 +1,146 @@
+use std::borrow::Cow;
+
+use bezier_rs::BezierHandles;
+use glam::DVec2;
+use petgraph::graph::{EdgeIndex, NodeIndex, UnGraph};
+use rustc_hash::FxHashMap;
+
+use super::{PointId, SegmentId, VectorData};
+
+pub struct IndexedVectorData<'a> {
+	/// This structure owns the data to allow convenient modification + storage
+	/// Can work on owned data or a reference
+	pub(crate) data: Cow<'a, VectorData>,
+	/// Points and segments form a graph. Store it here in a form amenable to graph algos
+	pub(crate) point_graph: UnGraph<Point, Segment>,
```

Comment:
Not all fields are used here. They would be useful in other contexts - TODO consider whether to delete or not.